packages/editor/src/utils/media-upload/index.js

```diff
@@ -35,13 +35,13 @@ export default function mediaUpload( {
 	const wpAllowedMimeTypes = getEditorSettings().allowedMimeTypes;
 	maxUploadFileSize =
 		maxUploadFileSize || getEditorSettings().maxUploadFileSize;
-
+	const currentPostId = getCurrentPostId();
```

Comment:
I wonder what the post id is used for? Attachments?

It seems like the way this function works might lead to some issues. In the site editor an image could be uploaded in the context of a post if you were editing post content, but I'm not sure `getCurrentPostId` is guaranteed to return the right thing ... maybe there are some contexts where it returns the template id instead of the content post id (as per this bug).

I wonder if the post id should really be passed into `mediaUpload` in the majority of cases, and maybe determined using `useEntityProp` within the block.

Reply:
When I edited a page in the context of the site editor with this patch the post id was set correctly in this method. But previously with the site editor specific method no post id was being set for the media upload even when editing in the context of the post/page, so it seems like moving to this shared method might have resolved an existing potential issue as well as causing the bug.

I don't know what the impact might be of all the images previously uploaded in pages in the site editor not having the post id set might be - so it could be worth looking further into why it is set. I will look at that tomorrow if @youknowriad doesn't know.

Reply:
> maybe there are some contexts where it returns the template id instead of the content post id (as per this bug).

No, it always returns the postId no matter the context.

Reply:
That said, when you open a template directly in the site editor, it contains the template id, when you open the navigation block, it contains its id, when you open a pattern or template part, same...

Reply:
I guess I was thinking of whether it's possible to open a template (where it's set to the template slug) and the edit something inside the `Content` block of a nested post, but that doesn't seem possible in my testing.

Reply:
>  so it could be worth looking further into why it is se

Original PR here:

- https://github.com/WordPress/gutenberg/pull/6231

Since that PR, `editorMediaUpload` has been renamed to `mediaUpload` and refactored a bit.

Here's the reason it seems:

> Once the image was uploaded, it was correctly assigned to its parent post

Reply:
Having said that, I'm not entirely sure sending the post id is required at all 🤷🏻 

But it's been there so long it's expected, at least by Jetpack. See: https://github.com/Automattic/wp-calypso/issues/85373#issuecomment-1869907797

Here's a PR that forces the numerical ID: 

- https://github.com/WordPress/gutenberg/pull/57388

Reply:
Yes, here we go. Right in front my eyes. The `post` arg is an integer according to the `WP_REST_Attachments_Controller`

https://github.com/WordPress/wordpress-develop/blob/d1faee936de9f08b0e496c86f7cd24dfda6739f6/src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php#L946-L946